mono/utils/mono-threads-windows.c

```diff
@@ -29,13 +29,12 @@ mono_threads_suspend_begin_async_suspend (MonoThreadInfo *info, gboolean interru
 	HANDLE handle;
 	DWORD result;
 
-	handle = OpenThread (THREAD_ALL_ACCESS, FALSE, id);
+	handle = info->native_handle;
```

Comment:
Does the OpenThread call need done? I forget my Win32 but doesn't it convert a pseudu handle to a real handle or something for later API calls?

Reply:
I use DuplicateHandle in mono_threads_suspend_register in order to convert a pseudohandle to a real handle. OpenThread is unnecessary.